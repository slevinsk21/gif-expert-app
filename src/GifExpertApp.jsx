// react
import { useState } from 'react';

// components
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch', 'DragÃ³n ball']);

  const onAddCategory = newCategory => {
    const newValueExist = categories.find(
      i => i.toLowerCase() === newCategory.toLowerCase()
    );
    // const newValueExist = categories.includes(newCategory);
    !newValueExist
      ? setCategories([newCategory, ...categories])
      : alert(`la cattegorÃ­a: ${newCategory} ya fue agregada...`);
  };

  const onDeleteCategory = categoryToDelete => {
    setCategories(categories.filter(category => category !== categoryToDelete));
  };

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory onAddCategory={onAddCategory} />

      {categories.length ? (
        categories.map(category => (
          <GifGrid
            key={category}
            category={category}
            onDeleteCategory={onDeleteCategory}
          />
        ))
      ) : (
        <h1
          className='fade-in'
          style={{
            display: 'flex',
            justifyContent: 'center',
            maxWidth: 380,
            lineHeight: '42px',
            margin: '50px auto',
            textAlign: 'center'
          }}
        >
          AÃºn no hay nada por aquÃ­ ðŸ˜…... por favor agrega un nombre
        </h1>
      )}
    </>
  );
};
