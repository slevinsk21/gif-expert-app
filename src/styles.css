/* globals */
* {
  font-family: 'Ubuntu', sans-serif;
}
html {
  background-color: #252525;
  color: #f3f3f3;
  padding: 12px;
}
input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}
h2 {
  font-size: 1.5rem;
}
h3 {
  font-size: 3rem;
  margin-bottom: 5px;
}

/* card */
.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}
.card {
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;
  max-width: 220px;
}
.card p {
  background-color: white;
  color: #333;
  flex: 1;
  margin-top: 5px;
  padding: 5px 20px 0px 20px;
  text-align: center;
  text-transform: capitalize;
}
.card img {
  width: 100%;
}

/* fade */
.fade-in {
  transition: all 0.6s;
  animation: fadeIn 0.6s;
  -webkit-animation: fadeIn 0.6s;
  -moz-animation: fadeIn 0.6s;
  -o-animation: fadeIn 0.6s;
  -ms-animation: fadeIn 0.6s;
}

@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@-moz-keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@-webkit-keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@-o-keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@-ms-keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
