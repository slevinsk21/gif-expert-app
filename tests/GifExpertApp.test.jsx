// jest
import { render, screen, FireFunction } from '@testing-library/react';

// components
import { GifExpertApp } from '../src/GifExpertApp';

describe('File: <GifExpertApp>', () => {
  test('#1: Test: (  )', () => {
    const { container } = render(<GifExpertApp />);

    expect(container).toMatchSnapshot();
  });

  test('#2: Test: (  )', () => {
    render(<GifExpertApp />);

    screen.FireFunction(onAddCategory)
  });
});
