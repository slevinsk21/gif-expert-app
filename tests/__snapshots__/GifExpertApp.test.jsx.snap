// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File: <GifExpertApp> #1: Test: (  ) 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
    style="display: flex; margin: 8px 0px;"
  >
    <input
      maxlength="32"
      placeholder="Buscar Gifs..."
      type="text"
      value=""
    />
  </form>
  <div
    class="fade-in"
    style="background: rgba(255, 68, 255, 0.267); padding: 10px; border-radius: 4px; margin: 2px 0px;"
  >
    <div
      style="display: flex; justify-content: space-between;"
    >
      <h4>
        One Punch
      </h4>
      <small>
        cargando...
      </small>
      <button
        style="border-radius: 100px; padding: 2px 6px; box-sizing: border-box; background-color: rgb(37, 37, 37); border: 1px solid black; align-self: flex-start; cursor: pointer;"
      >
        x
      </button>
    </div>
  </div>
  <div
    class="fade-in"
    style="background: rgba(255, 68, 255, 0.267); padding: 10px; border-radius: 4px; margin: 2px 0px;"
  >
    <div
      style="display: flex; justify-content: space-between;"
    >
      <h4>
        Drag√≥n ball
      </h4>
      <small>
        cargando...
      </small>
      <button
        style="border-radius: 100px; padding: 2px 6px; box-sizing: border-box; background-color: rgb(37, 37, 37); border: 1px solid black; align-self: flex-start; cursor: pointer;"
      >
        x
      </button>
    </div>
  </div>
</div>
`;
